apps/betterangels-backend/clients/types.py

```diff
@@ -104,10 +111,31 @@ def search(
 
         queryset = queryset.filter(reduce(and_, q_objects))
 
-        return (
-            queryset,
-            Q(),
-        )
+        return (queryset, Q())
+
+    @strawberry_django.filter_field
+    def search_client(
+        self,
+        queryset: QuerySet,
+        info: Info,
+        value: ClientSearchInput,
+        prefix: str,
+    ) -> Tuple[QuerySet[ClientProfile], Q]:
+        """
+        Returns client profiles with exact match on all provided search fields (case insensitive).
+        All search fields are optional.
+        """
+        filters = {}
+
+        user_fields = ["first_name", "middle_name", "last_name"]
+
+        for field in user_fields:
+            if field_value := getattr(value, field):
+                filters[f"user__{field}__iexact"] = field_value.strip()
+
+        queryset = queryset.filter(**filters)
```

Comment:
can I assume `queryset.filter` returns null set if filters are empty?

Reply:
opposite, it returns everything. i can make it return nothing

Reply:
I'd typically assume that a search with no valid inputs would either return an empty set or an error. But I guess i've seen it done both ways. Probably safer to return nothing though